src/cli.ts

```diff
@@ -116,25 +120,62 @@ async function calc(
   return result;
 }
 
-const measureBundle = parseBundle(path.resolve(program.measureBundle));
-
-let patientBundles: fhir4.Bundle[] = [];
-// data requirements/queryInfo doesn't care about patient bundles, so just leave as an empty array if we're using that report type
-if (program.outputType !== 'dataRequirements' && program.outputType !== 'queryInfo') {
-  // Since patient bundles are no longer a mandatory CLI option, we should check if we were given any before
-  if (!program.patientBundles && !program.patientIds) {
-    console.error(`Must provide either patient bundle or patient ids when output type is "${program.outputType}"`);
-    program.help();
-  }
-  if (program.patientBundles) {
-    if (program.patientIds) {
-      console.error('Cannot use both --patient-bundles and --patient-ids flags');
+async function populatePatientBundles() {
+  let patientBundles: fhir4.Bundle[] = [];
+  // data requirements/queryInfo doesn't care about patient bundles, so just leave as an empty array if we're using that report type
+  if (program.outputType !== 'dataRequirements' && program.outputType !== 'queryInfo') {
+    // Since patient bundles are no longer a mandatory CLI option, we should check if we were given any before
+    if (!program.patientBundles && !program.patientIds && !program.groupId) {
+      console.error(
+        `Must provide either patient bundle, patient ids, or group id when output type is "${program.outputType}"`
+      );
       program.help();
     }
-    patientBundles = program.patientBundles.map((bundlePath: string) => parseBundle(path.resolve(bundlePath)));
+    // patient bundles is provided -- should not use patientIds or groupId to populate patient bundles
+    if (program.patientBundles) {
+      if (program.patientIds) {
+        console.error('Cannot use both --patient-bundles and --patient-ids flags');
+        program.help();
+      }
+      if (program.groupId) {
+        console.error('Cannot use both --patient-bundles and --group-id flags');
+        program.help();
+      }
+      patientBundles = program.patientBundles.map((bundlePath: string) => parseBundle(path.resolve(bundlePath)));
+    }
+
+    // if we want to pass patient data into the fqm-execution API as a cql-exec-fhir patient source. Build patientSource
+    // from patientBundles and wipe patientBundles to be an empty array.
+    if (program.asPatientSource) {
+      let patientSource;
+      if (program.fhirServerUrl) {
+        if (!program.patientIds && !program.groupId) {
+          console.error(
+            'Must provide an array of patient ids with --patient-ids flag or group id with --group-id flag for calculation using AsyncPatientSource'
+          );
+        }
+        if (program.patientIds && program.groupId) {
+          console.error('Cannot provide both --patient-ids and --group-id flags.');
```

Comment:
I wonder if console.errors here is the best method for error reporting. The message quickly gets swallowed up by the output of the calculation, which can be unpredictable if the input params are wrong. Thoughts on throwing an error instead @mgramigna ? Could be a separate task also since this is not the only place we console.error in this file.

Reply:
Strategy that I've used in this file in the past is, in cases where we encounter an error that should halt execution of the cli entirely, use a combination of `console.error` and `program.help()`. The latter will print the usage of the cli to stdout and exit the process, making the user try again